File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longest = 0;
        HashSet<Integer> seen = new HashSet<>();
        for(int i=0;i<nums.length;i++){
            seen.add(nums[i]);
        }
        for(int num:seen){
            if(!seen.contains(num-1)){
                int count = 1;
                while(seen.contains(num+count)){
                    count++;
                }
                longest = Math.max(longest, count);
            }
        }
        return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        sets = set()
        for num in nums:
            sets.add(num)
        longest = 0
        for num in sets:
            if num-1 not in sets:
                length = 1
                while num+length in sets:
                    length += 1
                longest = max(longest, length)
        return longest